src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -2,23 +2,109 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 
 namespace Microsoft.Extensions.Logging.Console
 {
     public class ConsoleLoggerProvider : ILoggerProvider
     {
+        private readonly ConcurrentDictionary<string, ConsoleLogger> _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
+
         private readonly Func<string, LogLevel, bool> _filter;
-        private readonly bool _includeScopes;
+        private readonly IConsoleLoggerSettings _settings;
 
         public ConsoleLoggerProvider(Func<string, LogLevel, bool> filter, bool includeScopes)
         {
+            if (filter == null)
+            {
+                throw new ArgumentNullException(nameof(filter));
+            }
+
             _filter = filter;
-            _includeScopes = includeScopes;
+            _settings = new ConsoleLoggerSettings()
+            {
+                IncludeScopes = includeScopes,
+            };
+        }
+
+        public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            _settings = settings;
+
+            if (_settings.ChangeToken != null)
+            {
+                _settings.ChangeToken.RegisterChangeCallback(OnConfigurationReload, null);
+            }
+        }
+
+        private void OnConfigurationReload(object state)
```

Comment:
nit: rename to `OnSettingsReload` (for my own understanding: as in this view there is no Configuration but just settings...which could come from anywhere where our Configuration system is one of them..)